lib/chess/store/game.ex

```diff
@@ -41,6 +47,20 @@ defmodule Chess.Store.Game do
       or_where: game.opponent_id == ^user_id
   end
 
+  def validate_king_in_check(changeset, %Game{turn: turn}, %{board: board}) do
+    case Board.king_in_check?(board, turn) do
+      true ->
+        changeset
+        |> add_error(
+          :board,
+          "That move would result in your king being in check"
+        )
+      _ ->
+        changeset
+    end
```

Comment:
If you're asking a boolean, I think it's clearer to use `if Board.king_in_check?..` instead of `case`.
Also, just curious, does this account for _other_ players moving their piece to put their opponent's king in check?

Reply:
This only checks to see if a move will put the current player in check, as that would make it an invalid move. Putting the opponent in check is a valid move. I will need to check that at some point though for sure!